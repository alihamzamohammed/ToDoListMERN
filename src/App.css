.body {
  width: 95%;
  margin: 0 auto;
  margin-top: 1%;
  text-decoration: none;
}

/* .container {
  display: none;
} */

.title {
  text-align: center;
  padding: 1.5%;
}

.card-text {
  font-size: 12pt;
  font-weight: 400;
  margin-bottom: 0;
  white-space: pre-line;
}

.card-title {
  padding: 1%;
  padding-top: 1.25rem;
}

.category {
  border: 1px solid #ddd;
  padding: 1.5%;
  margin: 1%;
  padding-bottom: 2%;
}

.category-header {
  padding: 1%;
  padding-bottom: 2%;
}

.category-name {
  vertical-align: middle;
  text-align: center;
}

.category-id {
  color: #777;
  font-size: 17pt;
  text-align: left;
  vertical-align: middle;
}

.todo-item {
  margin: 0 auto;
  text-align: center;
  width: 18rem;
}

.todo-container {
  padding: 1.25%;
}

.todo-id {
  text-align: centre;
  z-index: 100;
  position: absolute;
  top: 12px;
  left: 50%;
  transform: translate(-50%, 0);
  /* left: 12px; */
}

.spinner {
  margin: 5%;
  text-align: center;
}

.spinner-border {
  width: 5rem;
  height: 5rem;
}

.spinner p {
  padding: 1%;
  font-family: "Segoe UI", Tahoma, Geneva, Verdana, sans-serif;
  font-size: 19pt;
  font-weight: 350;
}

.text-input {
  width: 75%;
  margin: 0 auto;
  padding: 2%;
}

.form-button {
  margin-right: 1%;
}

.response {
  text-align: center;
  color: green;
}

.edit-form {
  display: none;
}

.link-text {
  color: inherit;
  text-decoration: inherit;
}
