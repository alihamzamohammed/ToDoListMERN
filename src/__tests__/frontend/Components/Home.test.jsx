import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";

import { BrowserRouter as Router } from "react-router-dom";
import App from "../../../Components/Home";

describe("Home page tests", () => {
  let container;

  beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  it("Renders category with todos", async () => {
    const testCats = [
      {
        name: "Test Category",
        _id: "123",
        todos: [{ title: "test todo", contents: "test todo contents" }],
      },
    ];

    global.fetch = jest.fn().mockImplementation(() =>
      Promise.resolve({
        json: () => Promise.resolve(testCats),
      })
    );

    await act(async () => {
      render(
        <Router>
          <App></App>
        </Router>,
        container
      );
    });

    expect(container).toMatchSnapshot();
  });

  it("Renders only category", async () => {
    const testCats = [
      {
        name: "Test Category",
        _id: "123",
      },
    ];

    global.fetch = jest.fn().mockImplementation(() =>
      Promise.resolve({
        json: () => Promise.resolve(testCats),
      })
    );

    await act(async () => {
      render(
        <Router>
          <App></App>
        </Router>,
        container
      );
    });

    expect(container).toMatchSnapshot();
  });

  it("Before rendering", () => {
    act(() => {
      render(
        <Router>
          <App></App>
        </Router>,
        container
      );
    });
    expect(container).toMatchSnapshot();
  });
});
