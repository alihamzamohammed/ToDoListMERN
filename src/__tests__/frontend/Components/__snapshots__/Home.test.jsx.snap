// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page tests Before rendering 1`] = `
<div>
  <h1
    class="display-3 title"
  >
    Your To-Dos:
  </h1>
  <div
    class="body"
  >
    <div
      class="spinner"
    >
      <div
        class="spinner-border spinner-border"
      />
      <p>
        Loading...
      </p>
    </div>
  </div>
</div>
`;

exports[`Home page tests Renders category with todos 1`] = `
<div>
  <h1
    class="display-3 title"
  >
    Your To-Dos:
  </h1>
  <div
    class="body"
  >
    <a
      class="link-text"
      href="/edit/category/0"
    >
      <div
        class="category"
        id="category-0"
      >
        <div
          class="row"
        >
          <div
            class="category-id col"
            xmd="1"
          >
            <h4>
              0
            </h4>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="category-name col-12"
          >
            <h3>
              Unsorted Todos
            </h3>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="todo-container col"
          >
            <a
              class="link-text"
              href="/edit/todo/321"
            >
              <div
                class="todo-item card"
                id="todo-321"
              >
                <div
                  class="card-body"
                >
                  <div
                    class="card-title h5"
                  >
                    test todo
                  </div>
                  <p
                    class="card-text"
                  >
                    test todo contents
                  </p>
                  <div
                    class="text-muted todo-id card-subtitle h6"
                  >
                    321
                  </div>
                </div>
              </div>
            </a>
          </div>
        </div>
      </div>
    </a>
    <a
      class="link-text"
      href="/edit/category/123"
    >
      <div
        class="category"
        id="category-123"
      >
        <div
          class="row"
        >
          <div
            class="category-id col"
            xmd="1"
          >
            <h4>
              123
            </h4>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="category-name col-12"
          >
            <h3>
              Test Category
            </h3>
          </div>
        </div>
        <div
          class="row"
        />
      </div>
    </a>
  </div>
</div>
`;

exports[`Home page tests Renders only category 1`] = `
<div>
  <h1
    class="display-3 title"
  >
    Your To-Dos:
  </h1>
  <div
    class="body"
  >
    <a
      class="link-text"
      href="/edit/category/0"
    >
      <div
        class="category"
        id="category-0"
      >
        <div
          class="row"
        >
          <div
            class="category-id col"
            xmd="1"
          >
            <h4>
              0
            </h4>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="category-name col-12"
          >
            <h3>
              Unsorted Todos
            </h3>
          </div>
        </div>
        <div
          class="row"
        />
      </div>
    </a>
    <a
      class="link-text"
      href="/edit/category/123"
    >
      <div
        class="category"
        id="category-123"
      >
        <div
          class="row"
        >
          <div
            class="category-id col"
            xmd="1"
          >
            <h4>
              123
            </h4>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="category-name col-12"
          >
            <h3>
              Test Category
            </h3>
          </div>
        </div>
        <div
          class="row"
        />
      </div>
    </a>
  </div>
</div>
`;
